File: frontend/src/http/httpTypes.ts
export type SignupBody = {
  username: string;
  password: string;
};

export type SignupResp = {
  signupSuccess: boolean;
  username?: string;
  errorCode?: ErrorCode;
};

export type ErrorResp = {
  errorCode: number;
  errorMessage: string;
};

export type LoginBody = {
  username: string;
  password: string;
};

export type LoginResp = {
  loginSuccess: boolean;
  username?: string;
  errorCode?: ErrorCode;
};

export enum ErrorCode {
  UNKNOWN,
  BAD_JSON,
  USER_EXISTS,
  USER_DOES_NOT_EXIST,
  INCORRECT_PASSWORD,
}

File: auth-server/httpTypes.go
package main

type ErrorCode = int64
const (
	UNKNOWN ErrorCode = iota
	BAD_JSON
	USER_EXISTS
	USER_DOES_NOT_EXIST
	INCORRECT_PASSWORD
	TOKEN_EXPIRED
)

type LoginBody struct {
	Username string  `json:"username"`
	Password string   `json:"password"`
}

type LoginResp struct {
	LoginSuccess bool `json:"loginSuccess"`
	Username string `json:"username"`
	ErrorCode ErrorCode `json:"errorCode"`
}

type SignupBody struct {
	Username string  `json:"username"`
	Password string   `json:"password"`
}

type SignupResp struct {
	SignupSuccess bool `json:"signupSuccess"`
	Username string `json:"username"`
	ErrorCode ErrorCode `json:"errorCode"`
}

type ErrorResp struct {
	ErrorCode ErrorCode `json:"errorCode"`
	ErrorMesage string `json:"errorMessage"`
}

